penaltymodel_mip/penaltymodel/mip/generation.py

```diff
@@ -257,9 +243,13 @@ def _generate_ising(graph, table, decision, min_classical_gap, linear_energy_ran
             const = solver.Constraint(val, val)  # equality constraint
             const.SetCoefficient(var, 1)
 
-    objective = solver.Objective()
-    objective.SetCoefficient(gap, 1)
-    objective.SetMaximization()
+    if auxiliary or len(table) != 2*len(decision):
```

Comment:
I think this if-statement is asking about non-empty auxiliary or when the table is NOT fully specified. Thus, I believe it should be
```
2**len(decision)
```

Reply:
Merging and making this change.